engine/runtime-integration-tests/src/test/scala/org/enso/compiler/pass/analyse/alias/graph/test/AliasAnalysisTest.scala

```diff
@@ -203,11 +209,22 @@ class AliasAnalysisTest extends CompilerTest {
     }
 
     "be able to check if a provided scope is a child of the current scope" in {
-      child1.toScope().isChildOf(complexScope) shouldEqual true
-      child2.toScope().isChildOf(complexScope) shouldEqual true
-      childOfChild.toScope().isChildOf(complexScope) shouldEqual true
-
-      complexScope.isChildOf(child1.toScope()) shouldEqual false
+      child1
+        .toScope()
+        .asInstanceOf[GraphImpl.Scope]
```

Comment:
Couldn't `Scope` be also refactored  so that it is part of `Graph` interface? All this casting is horrible and not very typesafe.

Reply:
That's because the `AliasAnalysisTest` tests internal details that aren't available via `Graph` and `Graph.Scope` (there already is such an API class) and one has to cast to the only implementation.